Framework/Kernel/src/InstrumentInfo.cpp

```diff
@@ -121,16 +121,62 @@ std::string InstrumentInfo::filePrefix(unsigned int runNumber) const {
 }
 
 /// Returns the name of the live listener
-const std::string &InstrumentInfo::liveListener() const {
-  return m_liveListener;
+std::string InstrumentInfo::liveListener(const std::string &name) const {
+  if (!hasLiveListenerInfo())
+    return "";
```

Comment:
Is there a reason that an empty string is being returned rather than throwing an exception?